src/plugins/bfetch/public/streaming/from_streaming_xhr.ts

```diff
@@ -41,14 +45,27 @@ export const fromStreamingXhr = (
 
   xhr.onprogress = processBatch;
 
+  const onBatchAbort = () => {
+    if (xhr.readyState !== 4) {
+      aborted = true;
+      xhr.abort();
+      subject.complete();
+      if (signal) signal.removeEventListener('abort', onBatchAbort);
+    }
+  };
+
+  if (signal) signal.addEventListener('abort', onBatchAbort);
+
   xhr.onreadystatechange = () => {
     // Older browsers don't support onprogress, so we need
     // to call this here, too. It's safe to call this multiple
     // times even for the same progress event.
     processBatch();
 
     // 4 is the magic number that means the request is done
-    if (xhr.readyState === 4) {
+    if (!aborted && xhr.readyState === 4) {
```

Comment:
I'm thinking maybe it is better to check for `aborted` at the very top of this function, so that we don't execute `processBatch()`.

```ts
  xhr.onreadystatechange = () => {
    if (aborted) return;

    // Older browsers don't support onprogress, so we need
    // to call this here, too. It's safe to call this multiple
    // times even for the same progress event.
    processBatch();

    // 4 is the magic number that means the request is done
    if (xhr.readyState === 4) {
```